istioctl/cmd/config.go

```diff
@@ -0,0 +1,140 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"fmt"
+	"io"
+	"sort"
+	"text/tabwriter"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+
+	"istio.io/istio/pilot/pkg/serviceregistry/kube/controller"
+	"istio.io/pkg/env"
+)
+
+var (
+	// settableFlags are the flags used to istioctl
+	settableFlags = map[string]interface{}{
+		"istioNamespace":      env.RegisterStringVar("ISTIOCTL_ISTIONAMESPACE", controller.IstioNamespace, "istioctl --istioNamespace override"),
```

Comment:
```suggestion
		"istioNamespace":      env.RegisterStringVar("ISTIOCTL_ISTIO_NAMESPACE", controller.IstioNamespace, "istioctl --istioNamespace override"),
```

Reply:
I like your version better.
I chose to implement this using an `_` for each `-` in _istioctl_ flags.  We have always used `--istioNamespace` as a flag.  It should have been `--istio-namespace` in my opinion, but we should have made that change in 0.2.
We could use Viper's "alias" feature to make ISTIOCTL_ISTIO_NAMESPACE an "alias" for ISTIOCTL_ISTIONAMESPACE in the future.  We could also choose something like ISTIO_NAMESPACE which doesn't "stutter".

Reply:
Got it. Thanks for the clarification